routers/api/v1/repo/issue_label.go

```diff
@@ -310,10 +311,20 @@ func prepareForReplaceOrAdd(ctx *context.APIContext, form api.IssueLabelsOption)
 		return
 	}
 
-	labels, err = models.GetLabelsByIDs(form.Labels)
-	if err != nil {
-		ctx.Error(http.StatusInternalServerError, "GetLabelsByIDs", err)
-		return
+	labels = make([]*models.Label, 0, len(form.Labels))
+	for _, q := range form.Labels {
+		label, exist, err := parseLabel(q, ctx.Repo.Repository.ID)
+		if err != nil {
+			ctx.Error(http.StatusInternalServerError, "GetIssueByIndex", err)
+			return nil, nil, err
+		}
+
+		if !exist {
+			ctx.NotFound()
+			return nil, nil, nil
+		}
+
+		labels = append(labels, label)
```

Comment:
So ... if the number of labels to get here is more than say 6 this is a substantial degrade in performance. 

It would be better to do two single multivalued queries.